lib/puppet/util/tcp_validator.rb

```diff
@@ -0,0 +1,36 @@
+require 'socket'
+require 'timeout'
+require 'ipaddr'
+
+module Puppet
+  module Util
+    class TcpValidator
```

Comment:
The puppet_x.rb suggests to use `puppetx/moduleowner` as namespace. `PuppetX::Puppetlabs::TcpValidator` would sound good to me.